[Konvoy] Publish 1.6.0-beta docs (#3190)
mesosphere/dcos-docs-site: pages/ksphere/konvoy/1.6.0-beta/addons/dependencies/index.md

---
layout: layout.pug
navigationTitle: Dependencies
title: Addon Dependencies
menuWeight: 7
excerpt: Understand addon dependencies and interactions
enterprise: false
---

<!-- markdownlint-disable MD018 -->

> When deploying Konvoy, addons are a integral part of the cluster's functionality. When deploying or troubleshooting addons, it helps to understand how addons interact and may require other addons as dependencies.

Reviewer: ```suggestion
When deploying Konvoy, addons play an integral part of the clusters functionality. When deploying or troubleshooting addons, it helps to understand how addons interact and may require other addons as dependencies.
```

Reviewer 2: @russurquhart1 this isn't grammatically correct.

> If an addon's dependency does not successfully deploy, the addon requiring that dependency does not successfully deploy. Addon dependencies are determined by their `.spec.requires` key value pair item.

Reviewer: ```suggestion
If an addon's dependency does not deploy, the addon requiring that dependency does not deploy. The addon dependencies `.spec.requires` key value pair item decides their dependencies.
```

> For example, the cert-manager addon is a dependency of traefik. If we take a look at the traefik `ClusterAddon` object, we see the following `.spec.requires` key value pair item:

Reviewer: ```suggestion
For example, traefix has the cert-manager addon as a dependency. The traefik `ClusterAddon` object has the following `.spec.requires` key value pair item:
```

```yaml
   - matchLabels:
        kubeaddons.mesosphere.io/name: cert-manager
```

As indicated by the key `matchLabels`, the addon looks for its dependencies using labels on objects.

> If we then view the `.metadata.labels` key value pair item of the cert-manager `ClusterAddon` object, we see the following label:

Reviewer: ```suggestion
The `.metadata.labels` key value pair item of the cert-manager `ClusterAddon` object, has the following label:
```

```yaml
   kubeaddons.mesosphere.io/name: cert-manager
```

In this example, if the cert-manager addon does not deploy, the traefik addon does not deploy.

> In addition to understanding why an addon does not deploy, these concepts help troubleshoot an addon not working properly. An addon dependency could not be healthy and cause issues. The table below lists Konvoy addons and their dependencies.

Reviewer: ```suggestion
These concepts help troubleshoot an addon not deploying or working. Addon dependencies can cause issues. The table below lists Konvoy addons and their dependencies.
```
